src/components/PasswordSetterBox.js

```diff
@@ -195,6 +201,8 @@ class PasswordSetterBox extends Nimiq.Observable {
             this._passwordInput.reset();
             this.$el.classList.add('repeat');
             this.fire(PasswordSetterBox.Events.ENTERED);
+            window.history.replaceState({ repeat: false }, 'Keyguard');
+            window.history.pushState({ repeat: true }, 'Keyguard');
```

Comment:
Using the history, it is much better now. Only issue I have is the name of the state property. I think we should namespace it more specificly, something like `isPasswordRepeat`